services/worker-manager/src/api.js

```diff
@@ -609,8 +609,10 @@ builder.declare({
   }
 
   let expires, workerConfig;
+  const secret = `${slug.nice()}${slug.nice()}`;
   try {
-    const reg = await provider.registerWorker({worker, workerPool, workerIdentityProof});
+    const encryptedSecret = this.db.encrypt({ value: Buffer.from(secret, 'utf8') });
+    const reg = await provider.registerWorker({worker, workerPool, workerIdentityProof, secret: encryptedSecret });
```

Comment:
This is a little awkward.  It does save calling the `updateWorker` db function twice, but requires that all of the providers know to write this secret to the DB.  It works as-is, but might be worth making the second call to `updateWorker` for the conceptual clarity of not having to have every provider handle this correctly (including the static provider..).